frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java

```diff
@@ -0,0 +1,209 @@
+import static com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils.getMacheteFilePath;
+
+import java.util.Collections;
+
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.ModalityUiUtil;
+import git4idea.branch.GitBrancher;
+import git4idea.config.GitConfigUtil;
+import git4idea.repo.GitRepository;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.BranchLayoutException;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.DeleteBranchOnSlideOutSuggestionDialog;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod(GitMacheteBundle.class)
+@CustomLog
+public class SlideOutBackgroundable extends Task.Backgroundable {
+
+  private final String branchToSlideOutName;
+  @Nullable
+  private final String currentBranchNameIfManaged;
+  private final BranchLayout branchLayout;
+
+  private final IBranchLayoutWriter branchLayoutWriter;
+
+  @Nullable
+  private final GitRepository selectedGitRepository;
+
+  @Nullable
+  private final BaseEnhancedGraphTable graphTable;
+
+  private final Project project;
+
+  public static final String DELETE_LOCAL_BRANCH_ON_SLIDE_OUT_GIT_CONFIG_KEY = "machete.slideOut.deleteLocalBranch";
+
+  public SlideOutBackgroundable(Project project, String title, String branchToSlideOutName,
+      @Nullable GitRepository selectedGitRepository,
+      @Nullable String currentBranchNameIfManaged,
+      BranchLayout branchLayout,
+      IBranchLayoutWriter branchLayoutWriter,
+      @Nullable BaseEnhancedGraphTable graphTable) {
+    super(project, title);
+    this.project = project;
+    this.branchToSlideOutName = branchToSlideOutName;
+    this.currentBranchNameIfManaged = currentBranchNameIfManaged;
+    this.branchLayout = branchLayout;
+    this.branchLayoutWriter = branchLayoutWriter;
+    this.selectedGitRepository = selectedGitRepository;
+    this.graphTable = graphTable;
+
+    LOG.debug(() -> "Entering: branchToSlideOut = ${branchToSlideOutName}");
+
+    LOG.debug("Refreshing repository state");
+
+  }
+
+  @Override
+  @UIThreadUnsafe
+  public void run(ProgressIndicator indicator) {
+    deleteBranchIfRequired(branchToSlideOutName);
```

Comment:
Hmmm the only thing that `SlideOutBackgroundable` does in `run` is `deleteBranchIfRequired` (and not "slide out")? _sum ting wong_ with naming probably 🤔